MandatoryTask/MandatoryTask/CandyGiftBasket.java

```diff
@@ -0,0 +1,110 @@
+package MandatoryTask;
+
+import java.util.*;
+
+public class CandyGiftBasket {
+    private List<Candy> candies;
+    private final int MAX_WEIGHT_IN_GRAMS = 500;
+    private final int MAX_CALORIFIC = 2500;
+    private final int MAX_CANDY_COUNT = 10;
+
+    public CandyGiftBasket() {
+        candies = new ArrayList<>();
+    }
```

Comment:
```suggestion
    private int maxWeightInGram;
    private int maxCalorific;
    private int maxCandyCount;
    private static final int DEFAULT_MAX_WEIGHT_IN_GRAMS = 500;
    private static final int DEFAULT_MAX_CALORIFIC = 2500;
    private static final int DEFAULT_MAX_CANDY_COUNT = 10;
    public CandyGiftBasket(int maxWeightInGram, int maxCalorific, int maxCandyCount) {
        this.candies = new ArrayList<>();
        this.maxWeightInGram = getValueOrDefault(maxWeightInGram, DEFAULT_MAX_WEIGHT_IN_GRAMS);
        this.maxCalorific = getValueOrDefault(maxCalorific, DEFAULT_MAX_CALORIFIC);
        this.maxCandyCount = getValueOrDefault(maxCandyCount, DEFAULT_MAX_CANDY_COUNT);
    }
    public CandyGiftBasket() {
        this(DEFAULT_MAX_WEIGHT_IN_GRAMS, DEFAULT_MAX_CALORIFIC, DEFAULT_MAX_CANDY_COUNT);
    }
    private int getValueOrDefault(int value, int defaultValue) {
        return value >= 0 ? value : defaultValue;
    }
```
краще зробити щось типу такого з валідацією, дефолтними значеннями і декількома конструкторами.
```